### PR TITLE
【代码迁移】任务39 : SGDR: Stochastic Gradient Descent with Warm Restarts

### PR DESCRIPTION
**Pull request by HelloDuoLA**
代码迁移任务39中的子任务1，issue链接为：https://gitee.com/mindspore/community/issues/IAPG9X，运行结果如下图：

### CODE REVIEW

**Comment by nanzhaogang** [comment]
需要用MindSpore来实现https://github.com/gurucharanmk/PyTorch_CosineAnnealingWithRestartsLR

**Comment by HelloDuoLA** [comment]
原代码是自定义了一个学习率调度器，Mindpsore中的自定义学习率好像是生成一个学习率列表传入优化器？

**Comment by nanzhaogang** [comment]
https://www.mindspore.cn/docs/zh-CN/r2.3.1/api_python/mindspore.nn.html#%E5%8A%A8%E6%80%81%E5%AD%A6%E4%B9%A0%E7%8E%87
可以参考这部分，不是只能传一个list给优化器

**Comment by HelloDuoLA** [comment]
运行了这部分代码，然后将lr打印出来就是一个列表哎，https://www.mindspore.cn/docs/zh-CN/r2.3.1/api_python/mindspore.nn.html#%E5%8A%A8%E6%80%81%E5%AD%A6%E4%B9%A0%E7%8E%87:~:text=%E5%BD%93%E5%89%8D%E5%AD%A6%E4%B9%A0%E7%8E%87%E3%80%82-,import,),-net

查看这个文档 https://www.mindspore.cn/docs/zh-CN/r2.3.1/api_python/nn/mindspore.nn.cosine_decay_lr.html#mindspore.nn.cosine_decay_lr:~:text=list%5Bfloat%5D%E3%80%82%E5%88%97%E8%A1%A8%E5%A4%A7%E5%B0%8F%E4%B8%BA%20total_step%E3%80%82
也是显示返回值是个列表

**Comment by nanzhaogang** [comment]
好的，先按照这种形式来吧
